### PR TITLE
add anchor to link the list of supported secrets

### PR DESCRIPTION
**Pull request by pberthonneau**
### Why:

The list of supported secret is a key reference that get regular updates, hence we need to link to it easily from external docs.  Unfortunately its closest anchor is for the section header far above. Then a link to [About secret scanning on GitHub Enterprise Server](https://docs.github.com/en/enterprise-server@3.2/code-security/secret-scanning/about-secret-scanning#about-secret-scanning-on-github-enterprise-server) doesn't show the list in a normal screen.

### What's being changed:

I have just added a title above the list of supported secrets, so that we can link to it.

Another solution could have been to add an HTML anchor, but I don't know if this is in your habits.

### Check off the following:

- [x] I have reviewed my changes in staging (look for "Automatically generated comment" and click **Modified** to view your latest changes).
- [x] For content changes, I have completed the [self-review checklist](https://github.com/github/docs/blob/main/contributing/self-review.md#self-review).

### Writer impact (This section is for GitHub staff members only):

- [ ] This pull request impacts the contribution experience
  - [ ] I have added the 'writer impact' label
  - [ ] I have added a description and/or a video demo of the changes below (e.g. a "before and after video")

### CODE REVIEW

**Comment by ramyaparimi** [comment]
@pberthonneau 
Thanks so much for opening a PR! I'll get this triaged for review :zap:

**Review by mchammer01 — DISMISSED** [review]
@pberthonneau 👋🏻 - thanks for contributing to our docs ✨ 
I am fine to add a heading so that you have an anchor to easily link to ⚡ 
I'm just going to make a couple of small tweaks so that the article and its structure are consistent for all our products.

File: content/code-security/secret-scanning/about-secret-scanning.md [hunk 1]
```diff
@@ -36,6 +36,8 @@ If your project communicates with an external service, you might use a token or
 
 When {% data variables.product.prodname_secret_scanning %} detects a set of credentials, we notify the service provider who issued the secret. The service provider validates the credential and then decides whether they should revoke the secret, issue a new secret, or reach out to you directly, which will depend on the associated risks to you or the service provider. For an overview of how we work with token-issuing partners, see "[Secret scanning partner program](/developers/overview/secret-scanning-partner-program)."
 
+### List of supported secrets
```
Comment (mchammer01) [inline content/code-security/secret-scanning/about-secret-scanning.md:39]:
```suggestion
### List of supported secrets for public repositories
```

File: content/code-security/secret-scanning/about-secret-scanning.md [hunk 2]
```diff
@@ -71,6 +73,11 @@ Repository administrators and organization owners can grant users and teams acce
 {% ifversion fpt or ghes > 3.0 or ghec %}
 To monitor results from {% data variables.product.prodname_secret_scanning %} across your private repositories or your organization, you can use the {% data variables.product.prodname_secret_scanning %} API. For more information about API endpoints, see "[{% data variables.product.prodname_secret_scanning_caps %}](/rest/reference/secret-scanning)."{% endif %}
 
+{% ifversion ghes or ghae %}
+## List of supported secrets{% else %}
+### List of supported secrets
```
Comment (mchammer01) [inline content/code-security/secret-scanning/about-secret-scanning.md:78]:
```suggestion
### List of supported secrets for private repositories
```

**Review by mchammer01 — COMMENTED** [review]
A couple more tweaks and this will be ready for merging 🤞🏻

**Review by Arsalsaeed — COMMENTED** [review]
.

**Review by mchammer01 — APPROVED** [review]
LGTM ✨

File: content/code-security/secret-scanning/about-secret-scanning.md (hunk 2 above)
Comment (samiramiri88) [reply content/code-security/secret-scanning/about-secret-scanning.md]: ok  just minutes

**Review by Arsalsaeed — COMMENTED** [review]
Readme

**Review by Arsalsaeed — COMMENTED** [review]
Readme

**Comment by pberthonneau** [comment]
Thanks a lot!

**Pull request merged**